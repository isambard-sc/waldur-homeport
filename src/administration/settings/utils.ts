import { titleCase } from '@waldur/core/utils';

export const getKeyTitle = (key) =>
  titleCase(key.toLowerCase().replaceAll('_', ' '));
