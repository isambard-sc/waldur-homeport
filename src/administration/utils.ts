import { translate } from '@waldur/i18n';

export const AnnouncementTypeOptions = [
  {
    label: translate('Information'),
    value: 'information',
  },
  {
    label: translate('Warning'),
    value: 'warning',
  },
  {
    label: translate('Danger'),
    value: 'danger',
  },
];
