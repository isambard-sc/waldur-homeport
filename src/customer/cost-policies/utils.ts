import { translate } from '@waldur/i18n';

export const getCostPolicyActionOptions = () => [
  {
    value: 'block_creation_of_new_resources',
    label: translate('Block creation of new resources'),
  },
  {
    value: 'block_modification_of_existing_resources',
    label: translate('Block modification of existing resources'),
  },
  {
    value: 'notify_organization_owners',
    label: translate('Notify organization owners'),
  },
  { value: 'notify_project_team', label: translate('Notify project team') },
  {
    value: 'terminate_resources',
    label: translate('Schedule termination of existing resources'),
  },
];
