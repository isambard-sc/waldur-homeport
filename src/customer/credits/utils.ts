export const serializeCustomerCredit = (formData) => ({
  ...formData,
  customer: formData.customer.url,
  offerings: formData.offerings
    ? formData.offerings.map((offering) => offering.url)
    : undefined,
});

export const getCreditInitialValues = (row) => ({
  value: row.value,
  end_date: row.end_date,
  minimal_consumption_logic: row.minimal_consumption_logic,
  expected_consumption: row.expected_consumption,
  grace_coefficient: row.grace_coefficient,
  apply_as_minimal_consumption: row.apply_as_minimal_consumption,
});
