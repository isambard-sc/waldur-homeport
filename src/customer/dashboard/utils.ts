import { useSelector } from 'react-redux';

import { BookingFilterStateOption } from '@waldur/booking/BookingStateFilter';
import { BOOKING_RESOURCES_TABLE } from '@waldur/booking/constants';
import { bookingFormSelector } from '@waldur/booking/store/selectors';
import { RootState } from '@waldur/store/reducers';
import {
  selectTablePagination,
  selectTableSorting,
} from '@waldur/table/selectors';

import { Legend } from './types';

export const bookingsFilterStateSelector = (
  state: RootState,
): BookingFilterStateOption[] => bookingFormSelector(state)?.state;

export const useBookingsCalendarProps = () => {
  const bookingsFilterState = useSelector(bookingsFilterStateSelector);
  const bookingsListCurrentPage = useSelector(
    (state: RootState) =>
      selectTablePagination(state, BOOKING_RESOURCES_TABLE)?.currentPage,
  );
  const bookingsListPageSize = useSelector(
    (state: RootState) =>
      selectTablePagination(state, BOOKING_RESOURCES_TABLE)?.pageSize,
  );
  const bookingsListSorting = useSelector((state: RootState) =>
    selectTableSorting(state, BOOKING_RESOURCES_TABLE),
  );
  return {
    bookingsFilterState,
    bookingsListCurrentPage,
    bookingsListPageSize,
    bookingsListSorting,
  };
};

// Refer to https://stackoverflow.com/a/15125941
export const getDistinctColorsFromEvents = (events): Legend[] => {
  const flags = [],
    output = [];
  for (const event of events) {
    if (flags[event.color]) continue;
    flags[event.color] = true;
    output.push({
      color: event.color,
      name: event.offering_name,
    });
  }
  return output;
};
