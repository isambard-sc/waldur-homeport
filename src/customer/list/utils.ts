import { DateTime } from 'luxon';

export const makeAccountingPeriods = (start: DateTime) => {
  let date = DateTime.now().startOf('month');
  const diff = date.diff(start, 'months');
  const choices = [];
  for (let i = 0; i <= diff.months; i++) {
    const month = date.month;
    const year = date.year;
    const label = date.toFormat('MMMM, yyyy');
    choices.push({
      label,
      value: { year, month, current: i === 0 },
    });
    date = date.minus({ months: 1 });
  }
  return choices;
};
