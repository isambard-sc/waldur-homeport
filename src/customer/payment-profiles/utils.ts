import { translate } from '@waldur/i18n';

export const getInitialValues = (props) => ({
  name: props.name,
  payment_type: props.payment_type,
  end_date: props.attributes.end_date,
  agreement_number: props.attributes.agreement_number,
  contract_sum: props.attributes.contract_sum,
});

export const getPaymentProfileTypeOptions = () => [
  { label: translate('Fixed-price contract'), value: 'fixed_price' },
  { label: translate('Monthly invoices'), value: 'invoices' },
  {
    label: translate('Payment gateways (monthly)'),
    value: 'payment_gw_monthly',
  },
];
