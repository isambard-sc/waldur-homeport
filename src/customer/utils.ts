import { useMemo } from 'react';

import { translate } from '@waldur/i18n';
import { useTabs } from '@waldur/navigation/context';

export const getCustomerItems = () => [
  {
    title: translate('Profile'),
    to: 'organization.dashboard',
  },
  {
    title: translate('Settings'),
    to: 'organization.manage',
  },
  {
    title: translate('Payments'),
    to: 'organization.payments',
  },
  {
    title: translate('My offerings'),
    to: 'marketplace-my-offerings',
  },
  {
    title: translate('Billing'),
    to: 'organization.billing',
  },
  {
    title: translate('Audit logs'),
    to: 'organization.details',
  },
];

export const useCustomerItems = () => {
  const tabs = useMemo(() => getCustomerItems(), []);
  useTabs(tabs);
};
