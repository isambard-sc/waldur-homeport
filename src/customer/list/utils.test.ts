import { DateTime } from 'luxon';
import { describe, beforeEach, vi, afterEach, it, expect } from 'vitest';

import { makeAccountingPeriods } from './utils';

describe('makeAccountingPeriods', () => {
  beforeEach(() => {
    vi.useFakeTimers();
  });

  afterEach(() => {
    vi.useRealTimers();
  });
  it('iterates through months', () => {
    vi.setSystemTime(
      DateTime.fromObject({ year: 2021, month: 10, day: 1 }).toJSDate(),
    );
    const options = makeAccountingPeriods(
      DateTime.fromObject({ year: 2021, month: 1, day: 1 }),
    );
    expect(options).toMatchSnapshot();
  });
});
