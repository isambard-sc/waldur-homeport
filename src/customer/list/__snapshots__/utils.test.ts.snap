// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`makeAccountingPeriods > iterates through months 1`] = `
[
  {
    "label": "October, 2021",
    "value": {
      "current": true,
      "month": 10,
      "year": 2021,
    },
  },
  {
    "label": "September, 2021",
    "value": {
      "current": false,
      "month": 9,
      "year": 2021,
    },
  },
  {
    "label": "August, 2021",
    "value": {
      "current": false,
      "month": 8,
      "year": 2021,
    },
  },
  {
    "label": "July, 2021",
    "value": {
      "current": false,
      "month": 7,
      "year": 2021,
    },
  },
  {
    "label": "June, 2021",
    "value": {
      "current": false,
      "month": 6,
      "year": 2021,
    },
  },
  {
    "label": "May, 2021",
    "value": {
      "current": false,
      "month": 5,
      "year": 2021,
    },
  },
  {
    "label": "April, 2021",
    "value": {
      "current": false,
      "month": 4,
      "year": 2021,
    },
  },
  {
    "label": "March, 2021",
    "value": {
      "current": false,
      "month": 3,
      "year": 2021,
    },
  },
  {
    "label": "February, 2021",
    "value": {
      "current": false,
      "month": 2,
      "year": 2021,
    },
  },
  {
    "label": "January, 2021",
    "value": {
      "current": false,
      "month": 1,
      "year": 2021,
    },
  },
]
`;
